### PR TITLE
1010: Filter by datasource, for path based groups only.

### PR DESCRIPTION
**Pull request by raman-bt**

### CODE REVIEW

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/DrawableDB.java [hunk 1]
```diff
@@ -708,13 +761,14 @@ private void insertOrUpdateFile(DrawableFile f, @Nonnull DrawableTransaction tr,
         try {
             // "INSERT OR IGNORE/ INTO drawable_files (path, name, created_time, modified_time, make, model, analyzed)"
```
Comment (millmanorama) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/DrawableDB.java]: this comment is out of date now

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java [hunk 2]
```diff
@@ -124,6 +125,9 @@
     private final HashSetManager hashSetManager = new HashSetManager();
     private final CategoryManager categoryManager = new CategoryManager(this);
     private final DrawableTagsManager tagsManager = new DrawableTagsManager(null);
+    
+    // RAMAN TBD: initialize this to 0
```
Comment (rcordovano) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java]: Please resolve this TBD and remove the comment.

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java [hunk 3]
```diff
@@ -549,6 +568,9 @@ public synchronized void setToolbar(Toolbar toolbar) {
         }
         this.toolbar = toolbar;
         thumbnailSize.bind(toolbar.thumbnailSizeProperty());
+        
+        // RAMAN TBD: bind filterByDataSourceId to the data source dropdown in the toolbar.
```
Comment (rcordovano) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java]: Please resolve this TBD and remove the comment.

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/DrawableDB.java [hunk 4]
```diff
@@ -1162,14 +1239,25 @@ private void closeStatements() throws SQLException {
     }
 
     private PreparedStatement getGroupStatment(DrawableAttribute<?> groupBy) {
+     
+        //
```
Comment (rcordovano) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/DrawableDB.java]: Empty comment?

**Pull request merged**